src/Engine/Serialization/LegacyImages.cpp

```diff
@@ -345,9 +345,15 @@ void serialize(const Party &src, Party_MM7 *dst) {
 
     // MM7 uses an array of size 10 here, but we only store 5 elements. So zero it first.
     dst->partyTimes.bountyHuntingNextGenerationTime.fill(0);
-    serialize(src.PartyTimes.bountyHunting_next_generation_time, &dst->partyTimes.bountyHuntingNextGenerationTime, 5);
+    serialize(src.PartyTimes.bountyHuntNextGenTime, &dst->partyTimes.bountyHuntingNextGenerationTime, 5);
+
+    // Initially was one array but was splitted in two to simplify access
+    dst->partyTimes.shopsNextGenerationTime0 = 0;
+    dst->partyTimes.shopsNextGenerationTime.fill(0); // Original size is 53 with 0 element unused
```

Comment:
I think you don't need the `fill` call here. `serialize` checks that sizes are equal.

The `fill` for `bountyHuntNextGenTime` above is needed because there array sizes are different --- we're writing 5 elements into an array of 10.